addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderThingMonitorHandler.java

```diff
@@ -624,265 +739,451 @@ public void updateStatus(ThingStatus status) {
+                interval = 1000;
+                break;
+            default:
+                return false;
+        }
+        return ((now - lastUpdate) > interval) ? true : false;
     }
 
     @Override
-    protected void onFetchData() {
+    protected void onFetchData(RefreshPriority cyclePriority) {
+        IZoneMinderConnectionHandler session = null;
+        IMonitorDataGeneral data = null;
 
-        IZoneMinderSession session = null;
+        boolean refreshChannels = false;
 
-        session = aquireSession();
-        try {
-            IZoneMinderMonitor monitorProxy = ZoneMinderFactory.getMonitorProxy(session, getZoneMinderId());
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
+            return;
+        }
 
-            IZoneMinderMonitorData data = null;
-            IZoneMinderDaemonStatus captureDaemon = null;
-            IZoneMinderDaemonStatus analysisDaemon = null;
-            IZoneMinderDaemonStatus frameDaemon = null;
+        RefreshPriority curRefreshPriority = RefreshPriority.DISABLED;
 
-            data = monitorProxy.getMonitorData();
-            logger.debug("{}: URL='{}' ResponseCode='{}' ResponseMessage='{}'", getLogIdentifier(),
-                    monitorProxy.getHttpUrl(), monitorProxy.getHttpResponseCode(),
-                    monitorProxy.getHttpResponseMessage());
+        if (forcedPriority == RefreshPriority.UNKNOWN) {
+            return;
+        }
 
-            captureDaemon = monitorProxy.getCaptureDaemonStatus();
-            logger.debug("{}: URL='{}' ResponseCode='{}' ResponseMessage='{}'", getLogIdentifier(),
-                    monitorProxy.getHttpUrl(), monitorProxy.getHttpResponseCode(),
-                    monitorProxy.getHttpResponseMessage());
+        if (forcedPriority == RefreshPriority.DISABLED) {
+            curRefreshPriority = cyclePriority;
+        } else {
+            curRefreshPriority = forcedPriority;
+            forcedPriority = RefreshPriority.DISABLED;
+        }
 
-            analysisDaemon = monitorProxy.getAnalysisDaemonStatus();
-            logger.debug("{}: URL='{}' ResponseCode='{}' ResponseMessage='{}'", getLogIdentifier(),
-                    monitorProxy.getHttpUrl(), monitorProxy.getHttpResponseCode(),
-                    monitorProxy.getHttpResponseMessage());
+        session = null;
+        try {
+            session = aquireSession();
+            if (session == null) {
+                logger.warn("{}: Failed to aquire session for refresh, refresh loop for monitor will be skipped.",
+                        getLogIdentifier());
+                return;
+            }
 
-            frameDaemon = monitorProxy.getFrameDaemonStatus();
-            logger.debug("{}: URL='{}' ResponseCode='{}' ResponseMessage='{}'", getLogIdentifier(),
-                    monitorProxy.getHttpUrl(), monitorProxy.getHttpResponseCode(),
-                    monitorProxy.getHttpResponseMessage());
+        } catch (Exception ex) {
```

Comment:
Please catch specific exceptions instead of catching all exceptions. That way it is more clear what exceptions need to be handled and how this is done.